### ISSUE TITLE
Potential bug in the loss function

### ISSUE THREAD

**Issue by yygrechka** [ljmiao/PENCIL#5]
Why is Le being subtracted in the definition of Lc?
Is this a bug or just some clever optimization?

https://github.com/ljmiao/PENCIL/blob/a1b65d2ddca9a26c87a75c539575f438d32ba184/pencil_train.py#L244

**Comment by MohammedAlkhrashi** [comment]
I think this is the definition of the KL Divergence loss function: CrossEntropy (x,y)- Entropy(y), which is, I believe, what the author of the code is trying to do here
